lib/runtime-core/src/memory/ptr.rs

```diff
@@ -137,6 +137,17 @@ impl<T: Copy + ValueType> WasmPtr<T, Array> {
         let slice: &[u8] = unsafe { std::slice::from_raw_parts(ptr, str_len as usize) };
         std::str::from_utf8(slice).ok()
     }
+
+    /// Get a UTF-8 string representation of this `WasmPtr`, where the string is nul-terminated.
+    /// Note that this does not account for UTF-8 strings that _contain_ nul themselves,
+    /// [`get_utf8_string`] has to be used for those.
+    pub fn get_utf8_string_nul_terminated<'a>(self, memory: &'a Memory) -> Option<&'a str> {
```

Comment:
Can we change the name of this to `get_utf8_string_with_nul` to follow the naming conventions of the methods on CStr?